core/includes/config.inc

```diff
@@ -454,7 +454,12 @@ class Config {
 
     $this->name = $name;
     $this->storage = $storage;
-    $this->overrides = isset($config[$name]) ? $config[$name] : NULL;
+    if (is_array($config) || $config instanceof ArrayAccess) {
+      $this->overrides = isset($config[$name]) ? $config[$name] : NULL;
+    }
+    elseif (is_object($config)) {
```

Comment:
if $config is an object we should probably just ignore it.

Reply:
ok have removed that now @herbdool